backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -97,23 +98,34 @@
                 schema = @Schema(type = SchemaType.INTEGER)
             )
     })
-    public Uni<EndpointPage> getEndpoints(@Context SecurityContext sec, @BeanParam Query query, @QueryParam("type") List<String> targetType, @QueryParam("active") Boolean activeOnly) {
+    public Uni<EndpointPage> getEndpoints(
+            @Context SecurityContext sec,
+            @BeanParam Query query,
+            @QueryParam("type") List<String> targetType,
+            @QueryParam("active") Boolean activeOnly) {
         RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
 
         return sessionFactory.withSession(session -> {
             Uni<List<Endpoint>> endpoints;
             Uni<Long> count;
 
             if (targetType != null && targetType.size() > 0) {
-                Set<EndpointType> endpointType;
+                Set<CompositeEndpointType> compositeType;
                 try {
-                    endpointType = targetType.stream().map(s -> EndpointType.valueOf(s.toUpperCase())).collect(Collectors.toSet());
+                    compositeType = targetType.stream().map(s -> {
+                        String[] pieces = s.split(":", 2);
```

Comment:
Before splitting, we could use a regex to validate that the target type is valid and throw a `BadRequestException` otherwise.

Reply:
It would fail below when doing `EndpointType.valueOf`, do you think we would still need to validate here?

Reply:
Updated the code so that the returned error is more meaningful to the user. It would return something like:
`Unknown endpoint type: [stuff:splunk]`